crates/e2e/tests/e2e/cow_amm.rs

```diff
@@ -346,4 +346,14 @@ async fn cow_amm(web3: Web3) {
     })
     .await
     .unwrap();
+
+    // Check that the CoW AMM product factory listener works and catches the new CoW
+    // AMM deployment events
+    let mut auctions = mock_solver.get_auctions();
+    assert_eq!(auctions.len(), 1);
+    assert!(auctions
```

Comment:
Very nice that you already test this in the e2e test. 👍 
Just one thought on this. You don't really care about the solver only having received exactly 1 solution, right?
All that matters is that eventually the these addresses start showing up in the auction. So I would recommend looping over the auctions and asserting that at least one has these addresses.